source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -430,6 +434,66 @@ void InstanceImpl::PendingRequest::onFailure() {
 void InstanceImpl::PendingRequest::onRedirection(Common::Redis::RespValuePtr&& value,
                                                  const std::string& host_address,
                                                  bool ask_redirection) {
+  if (parent_.dns_cache_) {
+    resp_value_ = std::move(value);
+    ask_redirection_ = ask_redirection;
+    auto result =
+        parent_.dns_cache_->loadDnsCacheEntry(host_address, parent_.default_port_, false, *this);
+
+    cache_load_handle_ = std::move(result.handle_);
+    switch (result.status_) {
+    case Extensions::Common::DynamicForwardProxy::DnsCache::LoadDnsCacheEntryStatus::InCache: {
+      ASSERT(cache_load_handle_ == nullptr);
+      if (!result.host_info_.has_value() || !result.host_info_.value()->address()) {
+        auto host = host_;
+        onResponse(std::move(resp_value_));
+        host->cluster().stats().upstream_internal_redirect_failed_total_.inc();
+      } else {
+        doRedirection(std::move(resp_value_),
+                      formatAddress(result.host_info_.value()->address()->ip()), ask_redirection_);
+      }
+      return;
+    }
+    case Extensions::Common::DynamicForwardProxy::DnsCache::LoadDnsCacheEntryStatus::Loading:
+      ASSERT(cache_load_handle_ != nullptr);
+      return;
+    case Extensions::Common::DynamicForwardProxy::DnsCache::LoadDnsCacheEntryStatus::Overflow:
+      ASSERT(cache_load_handle_ == nullptr);
+      auto host = host_;
+      onResponse(std::move(resp_value_));
+      host->cluster().stats().upstream_internal_redirect_failed_total_.inc();
+      return;
+    }
+    PANIC_DUE_TO_CORRUPT_ENUM;
+  }
+
+  doRedirection(std::move(value), host_address, ask_redirection);
+}
+
+std::string InstanceImpl::PendingRequest::formatAddress(const Envoy::Network::Address::Ip* ip) {
```

Comment:
In practice I think this has to be non-null? Can you pass by reference?

Reply:
Yeah as far as I know Redis can't be used on Unix domain sockets and such.

Reply:
Or at least, typically isn't used... less so for cluster mode.